src/plugins/files/server/file_service/internal_file_service.ts

```diff
@@ -80,10 +81,33 @@ export class InternalFileService {
     }
   }
 
+  private async bulkGet(ids: string[]): Promise<IFile[]> {
+    try {
+      const metadatas = await this.metadataClient.bulkGet({ ids });
+      const result = metadatas.map(({ id, metadata }) => {
+        if (metadata.Status === 'DELETED') {
+          throw new FileNotFoundError('File has been deleted');
+        }
+        return this.toFile(id, metadata, metadata.FileKind);
+      });
+      return result;
+    } catch (e) {
+      if (SavedObjectsErrorHelpers.isNotFoundError(e)) {
```

Comment:
Will the saved object client throw an error if one of the files does not exist? I think it will just mark that item in the array with the `error` field.

Reply:
Yes you are right, that's how it is returned from the saved object `bulkGet`

```ts
if (docNotFound) {
  return {
    id,
    type,
    error: errorContent(SavedObjectsErrorHelpers.createGenericNotFoundError(type, id)),
  } as any as SavedObject<T>;
}
```

In https://github.com/elastic/kibana/pull/155636/commits/b97a0f1b1c0ac6e88a297379cc9b30f4a2465983 I removed this condition and made sure we return `null` in case of an error when fetching one of the saved object.